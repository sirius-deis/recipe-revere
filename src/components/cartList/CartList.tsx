import { FC, ReactNode } from "react";
import Cart from "../cart/Cart";
import List from "../list/List";

interface CartListProps {
  carts: Array<{
    title: string;
    icon: ReactNode;
    url: string;
  }>;
}

const CartList: FC<CartListProps> = ({ carts }) => {
  return (
    <div>
      <List>
        {carts.map((cart, i) => (
          <Cart key={i} {...cart} />
        ))}
      </List>
    </div>
  );
};

export default CartList;
