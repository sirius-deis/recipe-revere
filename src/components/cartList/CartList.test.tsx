import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CartList from "./CartList";
import { FaLeaf, FaSeedling } from "react-icons/fa";

const carts = [
  {
    title: "Vegetarian",
    icon: <FaLeaf />,
    url: "/search-results?category=vegetarian",
  },
  {
    title: "Vegan",
    icon: <FaSeedling />,
    url: "/search-results?category=vegan",
  },
];

describe("CartList component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <CartList carts={carts} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render component with correct titles", () => {
    render(
      <BrowserRouter>
        <CartList carts={carts} />
      </BrowserRouter>
    );
    expect(screen.getByText("Vegetarian")).toBeInTheDocument();
    expect(screen.getByText("Vegan")).toBeInTheDocument();
  });
  it("should render component with correct urls", () => {
    render(
      <BrowserRouter>
        <CartList carts={carts} />
      </BrowserRouter>
    );
    expect(screen.getByText("Vegetarian").closest("a")).toHaveAttribute(
      "href",
      "/search-results?category=vegetarian"
    );
    expect(screen.getByText("Vegan").closest("a")).toHaveAttribute(
      "href",
      "/search-results?category=vegan"
    );
  });
});
