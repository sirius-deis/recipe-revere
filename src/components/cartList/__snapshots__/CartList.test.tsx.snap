// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList component should match snapshot 1`] = `
<div>
  <div>
    <ul
      class="list vertical"
    >
      <li
        class="list-item"
      >
        <a
          href="/search-results?category=vegetarian"
        >
          <article
            class="cart"
          >
            <div
              class="icon"
              data-testid="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M546.2 9.7c-5.6-12.5-21.6-13-28.3-1.2C486.9 62.4 431.4 96 368 96h-80C182 96 96 182 96 288c0 7 .8 13.7 1.5 20.5C161.3 262.8 253.4 224 384 224c8.8 0 16 7.2 16 16s-7.2 16-16 16C132.6 256 26 410.1 2.4 468c-6.6 16.3 1.2 34.9 17.5 41.6 16.4 6.8 35-1.1 41.8-17.3 1.5-3.6 20.9-47.9 71.9-90.6 32.4 43.9 94 85.8 174.9 77.2C465.5 467.5 576 326.7 576 154.3c0-50.2-10.8-102.2-29.8-144.6z"
                />
              </svg>
            </div>
            <h3
              class="title"
            >
              Vegetarian
            </h3>
          </article>
        </a>
      </li>
      <li
        class="list-item"
      >
        <a
          href="/search-results?category=vegan"
        >
          <article
            class="cart"
          >
            <div
              class="icon"
              data-testid="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 96H0c0 123.7 100.3 224 224 224v144c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V320C288 196.3 187.7 96 64 96zm384-64c-84.2 0-157.4 46.5-195.7 115.2 27.7 30.2 48.2 66.9 59 107.6C424 243.1 512 147.9 512 32h-64z"
                />
              </svg>
            </div>
            <h3
              class="title"
            >
              Vegan
            </h3>
          </article>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
